konduit-serving-cli/src/main/java/ai/konduit/serving/cli/launcher/command/build/extension/ProfileCommand.java

```diff
@@ -0,0 +1,509 @@
+            return cudaInstallFromNvcc;
+        } else {
+            return findCudaInstallFromNvidiaSmi();
+        }
+    }
+
+    private static Pair<String, String> findCudaInstallFromNvcc() {
+        String mainCommand = "nvcc";
+
+        try {
+            String cudaVersion = findCudaVersion(Arrays.asList(mainCommand, "--version"), Pattern.compile("release (.*),"));
+            if(cudaVersion == null) {
+                return null;
+            } else {
+                return new Pair<>(cudaVersion, findCudaInstallPath(mainCommand, cudaVersion));
+            }
+        } catch (Exception exception) {
+            log.error("Couldn't find cuda version from {} command", mainCommand, exception);
+            System.exit(1);
+            return null;
+        }
+    }
+
+    private static Pair<String, String> findCudaInstallFromNvidiaSmi() {
+        String mainCommand = "nvidia-smi";
+
+        try {
+            String cudaVersion = findCudaVersion(Collections.singletonList(mainCommand), Pattern.compile("CUDA Version:\b+(.*)\b+"));
+            if(cudaVersion == null) {
+                return null;
+            } else {
+                return new Pair<>(cudaVersion, findCudaInstallPath(mainCommand, cudaVersion));
+            }
+        } catch (Exception exception) {
+            log.error("Couldn't find cuda version from {} command", mainCommand, exception);
+            System.exit(1);
+            return null;
+        }
+    }
+
+    private static String findCudaVersion(List<String> command,  Pattern pattern) throws IOException {
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(new ProcessBuilder(command).start().getInputStream()))) {
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                Matcher matcher = pattern.matcher(line);
+                if(matcher.find()) {
+                    return matcher.group(1).trim();
+                }
+                line = bufferedReader.readLine();
+            }
+        }
+
+        return null;
+    }
+
+    private static String findCudaInstallPath(String mainCommandName, String cudaVersion) throws IOException {
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(new ProcessBuilder(Arrays.asList("where", mainCommandName)).start().getInputStream()))) {
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                if(line.contains(cudaVersion)) {
+                    File parentFile = new File(line.trim()).getParentFile().getParentFile(); // to go back from <cuda_install_path>/bin/nvcc to <cuda_install_path>
+                    if(parentFile.exists()) {
+                        return parentFile.getAbsolutePath();
+                    }
+                    break;
+                }
+                line = bufferedReader.readLine();
+            }
+        }
+
+        return null;
+    }
```

Comment:
Another thing we can do is check the default CUDA install locations too. But if it's not on the path it may not be loadable anyway... So maybe we look at that in a separate PR (it'l take a bit of testing to ensure it works reliably)

Reply:
Issue created here: https://github.com/KonduitAI/konduit-serving/issues/436